src/realtime/realtime-hapi-plugin.ts

```diff
@@ -112,202 +52,87 @@ export class RealtimeHapiPlugin {
-        parents: event.payload.parents.map(item => toApiCommitId(projectId, item.id)),
-        branch,
-        project: String(projectId),
-      };
-      return this.toSSE(event, payload);
-
-    } catch (error) {
-      this.logger.error(`Unable to create StreamingCodePushedEvent`, { error, event });
-      throw Error('Unable to create StreamingCodePushedEvent');
+      const { teamId } = request.params;
+      const _teamId = parseInt(teamId, 10);
+      const predicate = (event: PersistedEvent<any>) => event.teamId === _teamId;
+      const stream = await this.getStream(_teamId, predicate, getLastEventId(request));
+      return this.streamReply(stream, request, reply);
+    } catch (err) {
+      this.logger.warn('Problems handling a SSE request', err);
+      return reply(Boom.wrap(err));
     }
   }
 
-  private getJsonApiModule() {
-    return this.jsonApiPlugin.getJsonApiModule();
-  }
-
-  private serializeApiEntity(type: string, entity: ApiEntity) {
-    return this.jsonApiPlugin.serializeApiEntity(type, entity);
-  }
-
-  private async commentAdded(event: Event<CommentAddedEvent>) {
-    const comment = await this.getJsonApiModule().toApiComment(event.payload);
-    const payload = this.serializeApiEntity('comment', comment).data;
-    return this.toSSE(event, payload);
-  }
-
-  private async commentDeleted(event: Event<CommentDeletedEvent>) {
-    const payload: StreamingCommentDeletedEvent = {
-      comment: String(event.payload.commentId),
-      teamId: event.payload.teamId,
-      deployment: toApiDeploymentId(event.payload.projectId, event.payload.deploymentId),
-    };
-    return this.toSSE(event, payload);
-  }
-
-  private async projectCreated(event: Event<ProjectCreatedEvent>) {
-    const payload: ApiProject = await this.jsonApiPlugin
-      .getEntity('project', api => api.getProject(event.payload.id));
-    return this.toSSE(event, payload);
-  }
-
-  private async deployment(event: Event<DeploymentEvent>) {
+  private async deploymentHandler(request: Hapi.Request, reply: Hapi.IReply) {
     try {
-      const deployment = event.payload.deployment;
-      const apiResponse = await this.jsonApiPlugin.getEntity(
-        'deployment', api => api.toApiDeployment(deployment.projectId, deployment));
-
-      const ssePayload: StreamingDeploymentEvent = {
-        teamId: event.payload.teamId,
-        branch: toApiBranchId(deployment.projectId, deployment.ref),
-        project: String(deployment.projectId),
-        commit: toApiCommitId(deployment.projectId, deployment.commitHash),
-        deployment: apiResponse.data,
-      };
-      return this.toSSE(event, ssePayload);
-    } catch (error) {
-      const msg = 'Could not convert DeploymentEvent to streaming event';
-      this.logger.error(msg, { error, event });
-      throw Error(msg);
+      const { projectId, deploymentId } = request.params;
+      const teamId = await request.getOpenTeamId();
```

Comment:
Doesn't this now assume that whenever we fetch events from this endpoint that they belong to the open team? This is not the case if I'm a logged in user from another team and I use a Deployment View as a landing page, at which point this endpoint will be used.